src/app/features/automated-status-update/automated-field-mapping/automated-field-mapping.component.ts

```diff
@@ -0,0 +1,193 @@
+import { Component, Inject, OnInit } from "@angular/core";
+import { FormControl, FormGroup } from "@angular/forms";
+import { MatButtonModule } from "@angular/material/button";
+import { MatOptionModule } from "@angular/material/core";
+import {
+  MAT_DIALOG_DATA,
+  MatDialogModule,
+  MatDialogRef,
+} from "@angular/material/dialog";
+import { MatFormFieldModule } from "@angular/material/form-field";
+import { MatOption, MatSelect } from "@angular/material/select";
+import { MatTooltipModule } from "@angular/material/tooltip";
+import { ConfigurableEnumDatatype } from "app/core/basic-datatypes/configurable-enum/configurable-enum-datatype/configurable-enum.datatype";
+import { ConfigurableEnumService } from "app/core/basic-datatypes/configurable-enum/configurable-enum.service";
+import { ConfigurableEnumValue } from "app/core/basic-datatypes/configurable-enum/configurable-enum.types";
+import { EntityFieldEditComponent } from "app/core/common-components/entity-field-edit/entity-field-edit.component";
+import { EntityFieldLabelComponent } from "app/core/common-components/entity-field-label/entity-field-label.component";
+import { EntityForm } from "app/core/common-components/entity-form/entity-form.service";
+import { FormFieldConfig } from "app/core/common-components/entity-form/FormConfig";
+import { Entity, EntityConstructor } from "app/core/entity/model/entity";
+import { EntitySchemaService } from "app/core/entity/schema/entity-schema.service";
+import { DialogCloseComponent } from "../../../core/common-components/dialog-close/dialog-close.component";
+
+/**
+ * Dialog to configure additional details for the "updated-from-referencing-entity"
+ * default value strategy, working in combination with the `AdminDefaultValueUpdatedComponent`.
+ */
+@Component({
+  selector: "app-automated-field-mapping",
+  imports: [
+    MatOptionModule,
+    MatSelect,
+    MatFormFieldModule,
+    MatOption,
+    MatDialogModule,
+    MatButtonModule,
+    MatTooltipModule,
+    EntityFieldEditComponent,
+    EntityFieldLabelComponent,
+    DialogCloseComponent,
+  ],
+  templateUrl: "./automated-field-mapping.component.html",
+  styleUrl: "./automated-field-mapping.component.scss",
+})
+export class AutomatedFieldMappingComponent implements OnInit {
+  /** The currently selected relatedReferenceField on the related entity */
+  selectedReferenceField: string;
+  /** all fields for selection as selectedReferenceField */
+  availableReferenceFields: string[];
+
+  /** The currently selected "relatedTriggerField" on the related entity */
+  selectedTriggerField: string | null = null;
+  /** all fields for selection as selectedTriggerField */
+  availableTriggerFields: { id: string; label: string; additional: string }[] =
+    [];
+
+  /**
+   *
+   */
+  mappingForms: {
+    [sourceId: string]: EntityForm<Entity>;
+  } = {};
+  selectedMappings: { [key: string]: any } = {};
+  sourceOptions: ConfigurableEnumValue[] = [];
```

Comment:
these are the individual trigger values of the selectedTriggerField, right?

Maybe we can find an even more descriptive variable name

Reply:
Yes, it's the values may be we can use this variable **triggerFieldValues**?